Source/Scene/Primitive.js

```diff
@@ -812,6 +820,11 @@ define([
                     this._state = PrimitiveState.COMBINING;
 
                     when(promise, function(packedResult) {
+                        if (!defined(packedResult)) {
+                            that._state = PrimitiveState.FAILED;
```

Comment:
> I think that might be useful so users won't have to check for undefined, but it'll be slower when creating new types arrays and it'll waste memory.

Since most of the time, geometry is valid, I think it's perfectly fine to always return an object (since when it's valid there's no slow down). The rare occasion that the geometry didn't get created isn't worth the pain of having users have to check for `undefined` the rest of the time. For example, all of the entity batching code for geometry assumes that a valid object will be returned if attributes were set; so I would have to update that if it suddenly starts returning undefined.  If we expected lots of failures to be the norm, then I would feel differently.